src/Composer/AddressSplitter.php

```diff
@@ -51,7 +51,7 @@ public function split(array $addresses): array
                     if ($typeCmp !== 0) {
                         return $typeCmp;
                     }
-                    if ($a->getType() === Address::TYPE_BYTE) {
+                    if ($a instanceof ByteReadRegisterAddress && $b instanceof ByteReadRegisterAddress) {
```

Comment:
`use ModbusTcpClient\Composer\Read\Register\ByteReadRegisterAddress;` is missing